kong/globalpatches.lua

```diff
@@ -521,6 +521,7 @@ return function(options)
 
     local client = package.loaded["kong.resty.dns.client"]
     if not client then
+      -- dns initialized here, can't be eliminated due to test dependencies
```

Comment:
Just a note: running code on module level is bad practice to begin with.

Reply:
We found that some balancer and DNS unit tests were run directly without running Kong.init(), but with globalpatch.lua loaded. Do you think it's a good approach to initialize Kong and then run these tests?

Reply:
> Just a note: running code on module level is bad practice to begin with.

Yes, actually we tried to remove running initialization logic in the `globalpatches` module, but that would make many test cases failure. I found that the logic of running initialization in `globalpatches` was added years ago and hard to eliminated now, that's why I add a comment here.